openassessment/assessment/api/staff.py

```diff
@@ -0,0 +1,292 @@
+    return True
+
+
+def get_score(submission_uuid, requirements):
+    """
+    Generate a score based on a completed assessment for the given submission.
+    If no assessment has been completed for this submission, this will return
+    None.
+
+    Args:
+        submission_uuid (str): The UUID for the submission to get a score for.
+        requirements (dict): Not used.
+
+    Returns:
+        A dictionary with the points earned, points possible,
+        contributing_assessments, and staff_id information.
+
+    """
+    assessment = get_latest_assessment(submission_uuid)
+    if not assessment:
+        return None
+
+    return {
+        "points_earned": assessment["points_earned"],
+        "points_possible": assessment["points_possible"],
+        "contributing_assessments": [assessment["id"]],
+        "staff_id": assessment["scorer_id"],
+    }
+
+
+def get_latest_assessment(submission_uuid):
+    """
+    Retrieve the latest staff assessment for a submission.
+
+    Args:
+        submission_uuid (str): The UUID of the submission being assessed.
+
+    Returns:
+        dict: The serialized assessment model
+        or None if no assessments are available
+
+    Raises:
+        StaffAssessmentInternalError
+
+    Example usage:
+
+    >>> get_latest_assessment('10df7db776686822e501b05f452dc1e4b9141fe5')
+    {
+        'points_earned': 6,
+        'points_possible': 12,
+        'scored_at': datetime.datetime(2014, 1, 29, 17, 14, 52, 649284 tzinfo=<UTC>),
+        'scorer': u"staff",
+        'feedback': u''
+    }
+
+    """
+    try:
+        assessments = Assessment.objects.filter(
+            submission_uuid=submission_uuid,
+            score_type=STAFF_TYPE,
+        )[:1]
+    except DatabaseError as ex:
+        msg = (
+            u"An error occurred while retrieving staff assessments "
+            u"for the submission with UUID {uuid}: {ex}"
+        ).format(uuid=submission_uuid, ex=ex)
+        logger.exception(msg)
+        raise StaffAssessmentInternalError(msg)
+
+    if len(assessments) > 0:
+        return full_assessment_dict(assessments[0])
+    else:
+        return None
+
+
+def get_assessment_scores_by_criteria(submission_uuid):
+    """Get the score for each rubric criterion
+
+    Args:
+        submission_uuid (str): The submission uuid is used to get the
+            assessment used to score this submission.
+
+    Returns:
+        (dict): A dictionary of rubric criterion names, with a score of
+            the staff assessments.
+
+    Raises:
+        StaffAssessmentInternalError: If any error occurs while retrieving
+            information from the scores, an error is raised.
+    """
+    try:
+        assessments = list(
+            Assessment.objects.filter(
+                score_type=STAFF_TYPE, submission_uuid=submission_uuid
+            )[:1]
+        )
+        scores = Assessment.scores_by_criterion(assessments)
+        return Assessment.get_median_score_dict(scores)
```

Comment:
Why does this need to get a median score, when the most recent staff score always wins? Is this just a shortcut because we know that only one score will be returned for each assessment? If not, it seems wrong to get the median score from two staff assessments. It would be good to either use a less confusingly named method, or have a comment to explain the assumptions being made.